website/project/model.py

```diff
@@ -2609,3 +2738,137 @@ def to_json(self):
                       for x in self.nodes if not x.is_deleted],
             "anonymous": self.anonymous
         }
+
+
+def validate_retraction_state(value):
+    acceptable_states = ['pending', 'retracted', 'cancelled']
+    if value not in acceptable_states:
+        raise ValidationValueError
+
+    return True
+
+
+class Retraction(StoredObject):
+    """Retraction object for public registrations."""
+
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+    justification = fields.StringField(default=None, validate=MaxLengthValidator(2048))
+    initiation_date = fields.DateTimeField(auto_now_add=datetime.datetime.utcnow)
+    initiated_by = fields.ForeignField('user', backref='retracted_by')
```

Comment:
"retracted" is probably a better backref key (and more consistent with other models). That way, you can get a user's retractions with `user.retraction__retracted`.